src/main/java/wedlog/address/model/person/GuestRsvpPredicate.java

```diff
@@ -1,28 +1,27 @@
 package wedlog.address.model.person;
 
-import java.util.List;
 import java.util.function.Predicate;
 
-import wedlog.address.commons.util.StringUtil;
 import wedlog.address.commons.util.ToStringBuilder;
 
 /**
- * Tests that a {@code Guest}'s {@code Rsvp} matches any of the keywords given.
+ * Tests that a {@code Guest}'s {@code Rsvp} matches the given input.
  */
 public class GuestRsvpPredicate implements Predicate<Guest> {
-    private final List<String> keywords;
+    private final String input;
 
     /**
      * Constructor for GuestRsvpPredicate.
      */
-    public GuestRsvpPredicate(List<String> keywords) {
-        this.keywords = keywords;
+    public GuestRsvpPredicate(String input) {
+        this.input = input;
     }
 
     @Override
     public boolean test(Guest guest) {
-        return keywords.stream()
-                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(guest.getRsvpStatus().value, keyword));
+        return input.isEmpty()
+                ? guest.getRsvpStatus().value.toLowerCase().contains("unknown") // check if value is unknown
+                : guest.getRsvpStatus().value.equalsIgnoreCase(input); // check if value is exactly same
     }
```

Comment:
`guest filter r/y` returns no guests (no error thrown). This seems rather unintuitive. Perhaps we can consider a few alternatives?
1) Throw an error for erroneous input in the parser
2) Accept partial matches for rsvp status here (but this means inputs such as `r/abcdefg` will not throw errors

<img width="805" alt="image" src="https://github.com/AY2324S1-CS2103T-F11-2/tp/assets/61283948/a36127d4-e9ab-4943-b878-ebeaf088278e">

Reply:
resolved!